cabal-install/main/Main.hs

```diff
@@ -445,40 +451,80 @@ reconfigureAction flags@(configFlags, _) _ globalFlags = do
     checkFlags [] globalFlags config
   pure ()
 
-buildAction :: (BuildFlags, BuildExFlags) -> [String] -> Action
-buildAction (buildFlags, buildExFlags) extraArgs globalFlags = do
-  let verbosity = fromFlagOrDefault normal (buildVerbosity buildFlags)
-      noAddSource = fromFlagOrDefault DontSkipAddSourceDepsCheck
-                    (buildOnly buildExFlags)
-  (useSandbox, config) <- loadConfigOrSandboxConfig verbosity globalFlags
-  distPref <- findSavedDistPref config (buildDistPref buildFlags)
-  -- Calls 'configureAction' to do the real work, so nothing special has to be
-  -- done to support sandboxes.
-  config' <-
-    reconfigure configureAction
-    verbosity distPref useSandbox noAddSource (buildNumJobs buildFlags)
-    mempty [] globalFlags config
-  nixShell verbosity distPref globalFlags config $ do
-    maybeWithSandboxDirOnSearchPath useSandbox $
-      build verbosity config' distPref buildFlags extraArgs
 
+buildAction :: (BuildFlags, BuildExFlags) -> [String] -> Action
+buildAction flags@(buildFlags, _) = buildActionForCommand
+  (Cabal.buildCommand defaultProgramDb)
+  verbosity
+  flags
+  where verbosity = fromFlagOrDefault normal (buildVerbosity buildFlags)
+
+showBuildInfoAction :: (BuildFlags, BuildExFlags) -> [String] -> Action
+showBuildInfoAction flags@(buildFlags, _) = buildActionForCommand
+  (Cabal.showBuildInfoCommand defaultProgramDb)
+  verbosity
+  flags
+    -- Default silent verbosity so as not to pollute json output
+  where verbosity = fromFlagOrDefault silent (buildVerbosity buildFlags)
```

Comment:
Wouldn't logging output go to stderr? This doesn't seem necessary to me.

Though I do vaguely remember new-build's `Up to date` message going to stdout maybe.

Reply:
Yeah, the up to date message goes to stdout. A little bit annoying.

Reply:
The stdout/stderr issue is a general issue (iirc we have a separate ticket for that) in cabal we don't need to solve in this PR :-)

Btw, I don't see a flag to redirect the output to a file. If there isn't we should have one (maybe `--buildinfo-json-output=FILE`)

Reply:
So, how should this be handled? 
What should be the value of the verbosity value?